src/transformers/kernels/falcon_mamba/selective_scan_with_ln_interface.py

```diff
@@ -0,0 +1,386 @@
+# coding=utf-8
+# Copyright 2024 Tri Dao, Albert Gu, Technological Innovation Institute and HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.nn.functional as F
+from torch.cuda.amp import custom_bwd, custom_fwd
+
+from einops import rearrange, repeat
+
+try:
+    import causal_conv1d_cuda
+except ImportError:
+    causal_conv1d_cuda = None
+
+import selective_scan_cuda
+from mamba_ssm.ops.triton.layer_norm import _layer_norm_fwd
```

Comment:
all these imports should be safe as this module is only used by FM only if `mamba-ssm` library is available. Einops, causal conv1d and selective scan cuda are automatically installed when `mamba-ssm` is installed